governance/approveDeny/approveDeny.groovy

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2021 JFrog Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import groovy.json.JsonSlurper
+import groovy.json.JsonOutput
+import groovyx.net.http.HTTPBuilder
+
+import org.apache.http.impl.client.AbstractHttpClient
+import org.apache.http.impl.client.DefaultHttpClient
+import org.apache.http.impl.conn.PoolingClientConnectionManager
+import org.apache.http.params.HttpParams
+
+import org.slf4j.Logger
+
+import static groovyx.net.http.ContentType.JSON
+import static groovyx.net.http.Method.POST
+
+
+def settings = new Settings(ctx, log)
+def compliance = new Compliance(log)
+
+def UNINITIALIZED = "null"
+
+download {
+    altResponse { request, responseRepoPath ->
+        try {
+            log.warn "Requesting artifact: $responseRepoPath"
+
+            def transaction_ip = request.clientAddress
+            def repository_name = responseRepoPath.getRepoKey()
+            def app_name = UNINITIALIZED
+
+            def user = security.currentUser()
+            def email = user.getEmail()
+
+            String[] items = responseRepoPath.getPath().split("/");
+            def product_name_space = UNINITIALIZED
+            int index = 0
+
+            if (items.length == 4) {
+                product_name_space = items[0]
+                index++
+            }
+
+            def module_name = items[index]
+            def version = items[index + 1]
+
+            if (items[index + 2].compareTo("manifest.json") == 0) {
+                log.warn "ipServer: ${settings.getIpServer()}"
+                log.warn "entitlementServer: ${settings.getEntitlementServer()}"
```

Comment:
Bring this to the debug log level when we update to "prod"

Reply:
Also, when we go prod, lets break out the "module extractor" as its own function, for ease of modification later.